Enforce maximum IRQL on BPF programs (#3149)
microsoft/ebpf-for-windows: docs/eBpfExtensions.md

structure from provided data and context buffers.
* `context_destroy`: Pointer to `ebpf_program_context_destroy_t` function that destroys a program type specific
context structure and populates the returned data and context buffers.

> * `required_irql`: IRQL that the eBPF program runs at.

Reviewer: Now that there's no minimum irql, does that mean that all eBPF programs might be invoked at PASSIVE level and hence are pre-emptible?  That doesn't seem right to me.

Author: I had considered having a minimum / maximum IRQL level for the program information, but wasn't sure if that made sense. Thoughts?

Reviewer: Before, `minimum_irql==maximum_irql` was implied by the name `required_irql`.  If they can be different, then yes we need two fields.  If they cannot be different, then `required_irql` is the correct name.

Author: Made min and max IRQL explicit parameters.

> * `maximum_irql`: The highest IRQL that the eBPF program can run at.

Reviewer: ```suggestion
* `maximum_irql`: The highest IRQL at which the eBPF program can be invoked.
```
For consistency with [ebpf_program_types.h](https://github.com/microsoft/ebpf-for-windows/pull/3149/files#diff-49fa4f21add58e3e7cee485187b8ec3aa7b84a480dcadd6d30792a78052a5998)
